### PR TITLE
Add GitHub Status Checks feature

### PR DESCRIPTION
**Pull request by gregmagolan**
Defaults to disabled as it is new and we don't want users to get it without opting in yet incase there are long tail bugs we didn't catch in our test cases

### CODE REVIEW

File: crates/aspect-cli/src/builtins/aspect/feature/github_status_checks.axl [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+            _state["_last_update_ms"] = now
+            github.status_checks.update(ctx, token, owner, repo, check_run_id, output = output)
+
+    lifecycle.task_started.append(_task_started)
+    lifecycle.task_update.append(_task_update)
+
+
+GithubStatusChecks = feature(
+    implementation = _github_status_checks,
+    enabled = False,
```
Comment (gregmagolan) [inline crates/aspect-cli/src/builtins/aspect/feature/github_status_checks.axl:333]: leaving disabled by default as it is new

File: crates/aspect-cli/src/builtins/aspect/lib/build_metadata.axl [hunk 2]
```diff
@@ -1,15 +1,95 @@
...
+        if slash < 0:
+            return {}
+        host = rest[:slash]
+        path = rest[slash + 1:]
+    if path.endswith(".git"):
+        path = path[:-len(".git")]
+    parts = path.split("/")
+    if len(parts) < 2:
+        return {"host": host}
+    return {"host": host, "owner": parts[0], "repo": parts[1]}
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply crates/aspect-cli/src/builtins/aspect/lib/build_metadata.axl]: Fix 1 — nested-namespace URLs ([build_metadata.axl:45-89](vscode-webview://1nm7sb8iuvcgk9m18gcgf9qsb7k5lf85et7bnamk3381q2ak7fb6/aspect-cli/crates/aspect-cli/src/builtins/aspect/lib/build_metadata.axl#L45-L89) and its callers)

_parse_git_remote_url now returns path (full namespace) alongside owner (top-level) and repo (leaf). Previously parts[1] was returned as repo, which silently truncated group/subgroup/project to owner=group, repo=subgroup.
_apply_vcs_override, _collect_buildkite, _collect_circleci, _collect_gitlab all now pass url_path = parsed.get("path") through _apply_repo_url, preserving the full namespace in REPO_URL.

File: crates/aspect-cli/src/builtins/aspect/lib/bazel_results.axl [hunk 3]
```diff
@@ -0,0 +1,3694 @@
...
+                results["executed_duration_ms"] += duration_ms
+
+        total_tracked = (len(results["failed_tests"]) +
+                         len(results["flaky_tests"]) +
+                         len(results["passed_tests"]))
+
+        if status in _FAILED_STATUSES:
+            results["failed_test_total"] += 1
+            if total_tracked < MAX_TEST_TARGETS:
+                results["failed_tests"].append({
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply crates/aspect-cli/src/builtins/aspect/lib/bazel_results.axl:674]: Fix 2 — non-pass test statuses ([bazel_results.axl:73-96](vscode-webview://1nm7sb8iuvcgk9m18gcgf9qsb7k5lf85et7bnamk3381q2ak7fb6/aspect-cli/crates/aspect-cli/src/builtins/aspect/lib/bazel_results.axl#L73-L96))

Added _TEST_STATUS_REMOTE_FAILURE and _TEST_STATUS_BLAZE_HALTED_BEFORE_TESTING to _FAILED_STATUSES. Both are non-pass terminal statuses in Bazel's BlazeTestStatus proto enum that were previously silently counted as passes.
Kept NO_STATUS and INCOMPLETE in the pass bucket with an explanatory comment — they're rare edge cases (target not run / test interrupted) that aren't strict passes or failures, and we lack a third bucket. This is the closest-available classification and is noted as such in the code.
